packages/core/src/common/path.ts

```diff
@@ -123,4 +123,24 @@ export class Path {
     toString(): string {
         return this.raw;
     }
+
+    relative(path: Path): Path | undefined {
+        if (!this.isEqualOrParent(path)) {
+            return undefined;
+        }
+        const relativePath = path.raw.substr(this.raw.length);
+        const raw = relativePath[0] === '/' ? relativePath.substr(1) : relativePath;
+        return new Path(raw);
+    }
+
+    isEqualOrParent(path: Path): boolean {
+        if (this.raw === path.raw) {
+            return true;
+        }
+        if (!this.raw || !path.raw) {
+            return false;
+        }
+        return path.raw.startsWith(this.raw);
```

Comment:
Still wondering about the path... Wouldn't `/home/me/test/file.txt` and `/home/me/t` cause problems?
In this case it would assume that `/home/me/t` is the parent of `/home/me/test/file.txt`, right?